starfish/core/imagestack/physical_coordinate_calculator.py

```diff
@@ -53,15 +51,19 @@ def recalculate_physical_coordinate_range(
     -------
     The new min and max physical coordinate values of the given dimension
     """
-    physical_pixel_size = _calculate_physical_pixel_size(coord_min, coord_max, dimension_size)
+    physical_pixel_size = _calculate_physical_pixel_size(coord_min, coord_max, dimension_size - 1)
     min_pixel_index = indexer if isinstance(indexer, int) else indexer.start
-    max_pixel_index = indexer if isinstance(indexer, int) else indexer.stop
-    # Add one to max pixel index to get end of pixel
-    max_pixel_index = max_pixel_index + 1 if max_pixel_index else dimension_size
-    new_min = _pixel_offset_to_physical_coordinate(
-        physical_pixel_size, min_pixel_index, coord_min, dimension_size)
-    new_max = _pixel_offset_to_physical_coordinate(
-        physical_pixel_size, max_pixel_index, coord_min, dimension_size)
+    if isinstance(indexer, int):
+        max_pixel_index = indexer
+    elif isinstance(indexer.stop, int):
```

Comment:
would this ever not be an int?

Reply:
I think if it's a range where the end is unbounded, it's a `None`.  In fact, that is not tested in my code. :(

Reply:
Updated with a test case for stop=`None`!